app/uk/gov/hmrc/gform/gform/AcknowledgementController.scala

```diff
@@ -120,18 +120,20 @@ class AcknowledgementController(
     val dateFormat = DateTimeFormatter.ofPattern("dd MMM yyyy")
     val formattedTime =
       s"""${submissionDetails.submittedDate.format(dateFormat)} ${submissionDetails.submittedDate.format(timeFormat)}"""
-
     val rows = List(
       cya_row("Submission date", formattedTime),
       cya_row("Submission reference", SubmissionRef(envelopeId).toString),
       cya_row("Submission mark", hashedValue)
     )
     val extraData = cya_section("Submission details", HtmlFormat.fill(rows)).toString()
-
     val declaration: List[(FormComponent, Seq[String])] = for {
-      formTemplateDecFields <- formTemplate.declarationSection.fields
-      formData              <- data.get(formTemplateDecFields.id)
-    } yield (formTemplateDecFields, formData)
+      formTemplateDecField <- formTemplate.declarationSection.fields
+      field <- formTemplateDecField.`type` match {
+                case g: Group => g.fields
```

Comment:
We are implementing this logic on couple of places already. We should make this into reusable public function and reuse it where possible.

Look at `Fields.flattenGroups` to start with. (There is also `def groupOrNot` in `StructuredFormDataBuilder` which should be replaced by our new function)

Reply:
Used the Fields.flattenGroups